### DISCUSSION TITLE
Minimum compiler version

### DISCUSSION THREAD

**Discussion by neheb** [Exiv2/exiv2#2573] [RFC]
current master supports GCC7 and above, which is default with Ubuntu 18.04 LTS.

@kmilos suggests bumping it to avoid workarounds. I think there's only one currently.

bumping the minimum to GCC9 (Ubuntu 20.04 LTS) allows using some C++20 features.

Anyone else have thoughts on this?

**Comment by kmilos** [comment]
I really don't think supporting more than 2 last LTS versions is necessary, nor wise. (If it were me, I'd only go for the single last LTS on main branch 😉)

As we require C++17 on the main branch, this was only _fully_ supported from GCC9.

This is also in line w/ the GHA runner image support: ["images are eventually deprecated according to our guidelines as we only support the latest 2 versions of an OS".](https://github.com/actions/runner-images)
